### PR TITLE
PHP 7.1 Notice: A non well formed numeric value encountered #12310

### PR DESCRIPTION
**Pull request by 810**
### Steps to reproduce the issue

PHP 7.1 RC3
Joomla 3..6.x
### Expected result

No notice message on joomla update page
### Actual result

Notice: A non well formed numeric value encountered in C:\wamp\www\develop\administrator\components\com_installer\models\warnings.php on line 45

### CODE REVIEW

**Comment by zero-24** [comment]
Please post your memory limit value.

**Comment by 810** [comment]
memory_limit 256M

File: administrator/components/com_installer/models/warnings.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ class InstallerModelWarnings extends JModelList
 	 */
 	public function return_bytes($val)
 	{
-		$val = trim($val);
+		$val = trim((int) $val);
```
Comment (andrepereiradasilva) [inline administrator/components/com_installer/models/warnings.php]: trim an integer value?

**Comment by 810** [comment]
i see also Low PHP memory limit. warning on the warning page.

But on the information page i see the correct values

**Comment by 810** [comment]
Ok, it should now work

**Comment by PhilETaylor** [comment]
I have this issue on a brand new iMac with PHP 7.1RC installed from brew 

System info
https://gist.github.com/PhilETaylor/3be36d59cd676fe36eda4f5520cae7a8

**Comment by PhilETaylor** [comment]
I have tested this item :white_check_mark: successfully on 53672692bbab2ab7600141d5dab2a6d1cc6a924c<br /><br />After applying path the warnings are removed. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12311">issues.joomla.org/tracker/joomla-cms/12311</a>.</sub>

**Comment by zero-24** [comment]
> After applying path the warnings are removed.

Just to be sure. I gues `path` is just a typo and should be `patch`?

**Comment by PhilETaylor** [comment]
yeah - should be after applying the patch, the warnings were removed :)

**Comment by PhilETaylor** [comment]
Still an issue in PHP 7.1.0RC5

**Comment by 810** [comment]
if someone else can test and approve, then we have a rtc

**Comment by PhilETaylor** [comment]
@mbabker do you have PHP 7.1 installed locally ? can you test this?

**Comment by 810** [comment]
any testers?

**Comment by 810** [comment]
Can we merge this one, now 7.1 is released.

**Comment by brianteeman** [comment]
@810 only if it is tested

**Comment by PhilETaylor** [comment]
I tested it :) and again today, still works ;-) 

I dunno who else uses PHP 7.1 in production, it seems we are a rare bread of people ;-)

**Comment by ralain** [comment]
I have tested this item :white_check_mark: successfully on 53672692bbab2ab7600141d5dab2a6d1cc6a924c<br /><br />Ran into the same issue. This fixes it. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12311">issues.joomla.org/tracker/joomla-cms/12311</a>.</sub>

**Comment by tkuschel** [comment]
I just reviewed the function, because i ran into the same issue with php 7.1 installed on an arch.
My comment:
The public function return_bytes($val) not only fixes the above issue, but the function was extended with new functionality and more costs.
It is a copy of http://php.net/manual/en/function.ini-get.php#106518 ! I'm not sure, whether this user example on php.net is the best for Joomla core.
a) Who needs the "gb" "kb" "mb" ? - Never tested.
b) What's about sending "  "?  - First, it jumps over the line `if(empty($val))return 0;`.
c) Expensive regular expression, aso.
But who cares? - the function is fixed.

**Comment by photodude** [comment]
Just set up a system with PHP 7.1 for a blog post, **glad to see that there is a fix for this issue**. 
seen when accessing administrator/index.php?option=com_joomlaupdate
```php
Notice: A non well formed numeric value encountered in \administrator\components\com_installer\models\warnings.php on line 45

Notice: A non well formed numeric value encountered in \administrator\components\com_installer\models\warnings.php on line 45

Notice: A non well formed numeric value encountered in \administrator\components\com_installer\models\warnings.php on line 45
```
Information about the system

| Setting | Value |
| --- | --- |
| PHP Built On | (Windows 7 Professional Edition Service Pack 1) AMD64 |
| Database Version | 5.7.17 |
| Database Collation | utf8_general_ci |
| Database Connection Collation | utf8mb4_general_ci |
| PHP Version | 7.1.2 |
| Web Server | Apache/2.4.20 (Win64) OpenSSL/1.0.2g PHP/7.1.2 |
| WebServer to PHP Interface | apache2handler |
| Joomla! Version | Joomla! 3.6.5 Stable [ Noether ] 1-December-2016 22:46 GMT |
| Joomla! Platform Version | Joomla Platform 13.1.0 Stable [ Curiosity ] 24-Apr-2013 00:00 GMT |
| User Agent | Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36 |

**Comment by PhilETaylor** [comment]
This is already fixed for the next release

**Comment by photodude** [comment]
Yes, glad to see that there is a fix for this issue.

**Comment by ittoufiq** [comment]
Hi, I want to know that, exactly which version fixed this issue. Thanks <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12311">issues.joomla.org/tracker/joomla-cms/12311</a>.</sub>

**Comment by zero-24** [comment]
@ittoufiq 3.7.0

**Pull request merged**